pf4j/src/main/java/org/pf4j/LegacyExtensionFinder.java

```diff
@@ -83,14 +83,21 @@ public LegacyExtensionFinder(PluginManager pluginManager) {
             Set<String> bucket = new HashSet<>();
 
             try {
-                URL url = ((PluginClassLoader) plugin.getPluginClassLoader()).findResource(getExtensionsResource());
-                if (url != null) {
-                    log.debug("Read '{}'", url.getFile());
-                    try (Reader reader = new InputStreamReader(url.openStream(), StandardCharsets.UTF_8)) {
-                        LegacyExtensionStorage.read(reader, bucket);
-                    }
-                } else {
+                Enumeration<URL> urls = ((PluginClassLoader) plugin.getPluginClassLoader()).findResources(getExtensionsResource());
+                if (urls == null || !urls.hasMoreElements()) {
```

Comment:
I think that `urls` cannot be null. The javadoc is not clear so we can keep null check.